.github/workflows/validate.yml

```diff
@@ -57,6 +57,20 @@ jobs:
 
       - uses: actions/checkout@v3
 
+      #  See the following link for a breakdown of the following step
+      #  https://github.com/haskell/actions/issues/7#issuecomment-745697160
+      #
+      # See https://github.com/haskell/cabal/pull/8739 for why Windows is excluded
+      - if: ${{ runner.os != 'Windows' }}
+        uses: actions/cache@v3
+        with:
+          # validate.sh uses a special build dir
+          path: |
+            ${{ steps.setup-haskell.outputs.cabal-store }}
```

Comment:
As GHAs are not prophetic, the variable `steps.setup-haskell.outputs.cabal-store` isn't set at this point, so you get the empty string, which probably means that `~/.cabal/store` isn't cached anymore.

Maybe `actionlint` would find such problems (haven't tried).  NB: I totally thrive on `actionlint`, best invention since sliced bread!  Made me so much more productive with workflow writing.

Reply:
Oh, so that may explain why the old GHCs seemed to have a wiped out cache and so started using XDG.

Reply:
So, I'm not certain what to do about it. Any ideas, good people? We wanted cache-restore before haskell/setup but we don't know where to restore to? I must confess I'm very hazy as to how actions/cache operates.

Reply:
Worst case, just repeat `cabal update` at the end of the cache restoration action and the bug with empty index should be resolved.

Reply:
But perhaps a better hack is to just touch ~/.cabal very early. That immediately turns on the backward compat mode of our XDG implementation and all is safe.

Reply:
Which means we could revert the commit that moves cache restoration around (it's already merged to master).

Reply:
> Worst case, just repeat `cabal update` at the end of the cache restoration action and the bug with empty index should be resolved.

@Mikolaj I don't quite get this suggestion: if cache restoration goes before `haskell/setup` there's no cabal to `cabal update`. Besides, inside the action there will be no ` ${{ steps.setup-haskell.outputs.cabal-store }}`, won't it? Do you propose replace it with explicit `~/.cabal/store` (or an XDG alternative if we decide to go with XDG).

Reply:
I meant that with the original late cache restoration, you can just repeat `cabal update` after cache is restored. That's, of course, wasteful.

Reply:
Oh, I got you. Well, considering all... that doesn't sound as bad.

Now I'm remembering that you can skip `cabal update` in the action (see `cabal-update` in the inputs table: https://github.com/haskell/actions/tree/main/setup#inputs).

But I'm starting to like the let's-go-XDG idea more and more.

Reply:
Note that the setup action does not take XDG into account when computing `outputs.cabal-store`:
- https://github.com/haskell/actions/issues/210

Reply:
So, eventually what I did is: I disabled cabal update during the action, and do update manually after cache restore. The action goes before the cache restore as before. Seems to work, and  I just didn’t have time to experiment with more interesting ideas like wiping the cache and XDGing everything. Could be a good idea to experiment with it in the future, actually, I open #8855.

Reply:
The setup action is now anti-XDG.